Add darwin subpackage

mzgoddard opened the pull request:
Add a package for publishing macos native files separate from the main package.

Manually releasing this package will involve a few simple steps:
1. Build and copy the `app` to the package.
2. Bump package version and publish to npm.
3. Update the version the main package depends on and publish it to npm.

mzgoddard commented:
I decided to make this change for macos on its own. I'll make a draft PR for the windows package. I decided on macos first since its my primary development environment.

Where should put releasing instructions for this package? We don't currently have instructions for publishing the package in the repo. We could put it in the README.

I wonder if there is tooling to help with publishing the dependency. But I'm not sure most common js tools for this will work with the added binary step?

-----

Adding this package I think will help with releasing our system specific built files. We can release the files for a system environment for an individual environment without having to build all target environments and collect them to one system.

mzgoddard commented:
I thought to put the package in a `packages/` directory first. But saw that it is gitignored because NuGet uses it.

jugglinmike reviewed (changes requested):
> Where should put releasing instructions for this package? We don't currently have instructions for publishing the package in the repo. We could put it in the README.

Let's add a `MAINTAINING.md` file and reference it from the `README.md`, kind of like what we're already doing with `CONTRIBUTING.md`.

> I wonder if there is tooling to help with publishing the dependency. But I'm not sure most common js tools for this will work with the added binary step?

I think we can get a long way with a minimal Bash script, but it's probably wise to stick with low-tech documentation until we have Windows support in place.

> Adding this package I think will help with releasing our system specific built files. We can release the files for a system environment for an individual environment without having to build all target environments and collect them to one system.

Yeah, that'll be nice!

> I thought to put the package in a `packages/` directory first. But saw that it is gitignored because NuGet uses it.

The name `dependencies` seems a little counter-intuitive to me. We don't have any plans to use NuGet, so let's remove that from `.gitignore` and go with `packages`. (I usually don't use such expansive `.gitignore` files; I honestly can't remember why I did so for this project.)

I'd prefer to keep the file structure agnostic of if/how we use NPM organizations. Would you remind removing the subdirectory named `@bocoup`?

mzgoddard commented:
@jugglinmike I rebased onto main and:

- Add a first pass of a `MAINTAINING.md` with steps for the darwin package. There are instructions for Windows but I'm not sure how accurate they are right now.
- Remove NuGet from `.gitignore`
- Renamed `dependencies/@bocoup/at-driver-generic-darwin-universal` to `packages/at-driver-generic-darwin-universal`

jugglinmike commented:
Thanks, @mzgoddard. The `MAINTAINING.md` file appears to be missing from the branch at the moment--maybe you forgot to `git add` it.

jugglinmike commented:
(This patch is in service of gh-29)
